core/services/ocr2/plugins/ocr2keeper/evm20/abi.go

```diff
@@ -0,0 +1,221 @@
+package evm
+
+import (
+	"fmt"
+	"math/big"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
+)
+
+type evmRegistryPackerV2_0 struct {
+	abi abi.ABI
+}
+
+// enum UpkeepFailureReason
+// https://github.com/smartcontractkit/chainlink/blob/d9dee8ea6af26bc82463510cb8786b951fa98585/contracts/src/v0.8/interfaces/AutomationRegistryInterface2_0.sol#L94
+const (
+	UPKEEP_FAILURE_REASON_NONE = iota
+	UPKEEP_FAILURE_REASON_UPKEEP_CANCELLED
+	UPKEEP_FAILURE_REASON_UPKEEP_PAUSED
+	UPKEEP_FAILURE_REASON_TARGET_CHECK_REVERTED
+	UPKEEP_FAILURE_REASON_UPKEEP_NOT_NEEDED
+	UPKEEP_FAILURE_REASON_PERFORM_DATA_EXCEEDS_LIMIT
+	UPKEEP_FAILURE_REASON_INSUFFICIENT_BALANCE
+)
+
+func NewEvmRegistryPackerV2_0(abi abi.ABI) *evmRegistryPackerV2_0 {
+	return &evmRegistryPackerV2_0{abi: abi}
+}
+
+func (rp *evmRegistryPackerV2_0) UnpackCheckResult(key ocr2keepers.UpkeepKey, raw string) (EVMAutomationUpkeepResult20, error) {
+	var (
+		result EVMAutomationUpkeepResult20
+	)
+
+	b, err := hexutil.Decode(raw)
+	if err != nil {
+		return result, err
+	}
+
+	out, err := rp.abi.Methods["checkUpkeep"].Outputs.UnpackValues(b)
+	if err != nil {
+		return result, fmt.Errorf("%w: unpack checkUpkeep return: %s", err, raw)
+	}
+
+	block, id, err := splitKey(key)
+	if err != nil {
+		return result, err
+	}
+
+	result = EVMAutomationUpkeepResult20{
+		Block:    uint32(block.Uint64()),
+		ID:       id,
+		Eligible: true,
+	}
+
+	upkeepNeeded := *abi.ConvertType(out[0], new(bool)).(*bool)
+	rawPerformData := *abi.ConvertType(out[1], new([]byte)).(*[]byte)
+	result.FailureReason = *abi.ConvertType(out[2], new(uint8)).(*uint8)
+	result.GasUsed = *abi.ConvertType(out[3], new(*big.Int)).(**big.Int)
+	result.FastGasWei = *abi.ConvertType(out[4], new(*big.Int)).(**big.Int)
+	result.LinkNative = *abi.ConvertType(out[5], new(*big.Int)).(**big.Int)
+
+	if !upkeepNeeded {
+		result.Eligible = false
+	}
+	// if NONE we expect the perform data. if TARGET_CHECK_REVERTED we will have the error data in the perform data used for off chain lookup
+	if result.FailureReason == UPKEEP_FAILURE_REASON_NONE || (result.FailureReason == UPKEEP_FAILURE_REASON_TARGET_CHECK_REVERTED && len(rawPerformData) > 0) {
+		var ret0 = new(performDataWrapper)
+		err = pdataABI.UnpackIntoInterface(ret0, "check", rawPerformData)
+		if err != nil {
+			return result, err
+		}
+
+		result.CheckBlockNumber = ret0.Result.CheckBlockNumber
+		result.CheckBlockHash = ret0.Result.CheckBlockhash
+		result.PerformData = ret0.Result.PerformData
+	}
+
+	// This is a default placeholder which is used since we do not get the execute gas
+	// from checkUpkeep result. This field is overwritten later from the execute gas
+	// we have for an upkeep in memory. TODO (AUTO-1482): Refactor this
+	result.ExecuteGas = 5_000_000
+
```

Comment:
I have a PR which returns the executeGas from `simulatePerformUpkeep` - do you need it returned from `checkUpkeep` also?